### PR TITLE
Ext: Added a sortBy feature

### LINKED ISSUE #1550: Extensibility: Details View: Table widget sorting/grouping by column
**Description**
I would like to be able to sort or group entries in the table widget by any column I choose so I can have a better view of the information presented.

Also make sure to handle the pagination on all table-like widgets (`Table`, `ResourceList` etc).

**Reasons**
Better user experience with big tables.

### PR DESCRIPTION
**Pull request by Lyczeq**
**Description**

Changes proposed in this pull request:

- [x] Added a sorting feature for the Table widget and the list view
- [x] Added a documentation for sorting
- [x] Added new screenshots for ResourceList and Table widgets
- [x] Added a sorting feature for the `ResourceLIst` and basic sorting for `ResourceRefs`
- [x] Modified examples folder to use sorting feature
- [x] Added a compareString function to jsonata
- [x] Fixed a resourceGraph, to reproduce it, remove the filter option in one of the dataSources that are used in the resourceGraph, the view should crash 💥 

**Related issue(s)**
Resolves #1550

### CODE REVIEW

File: docs/extensibility/display-widgets.md [hunk 1]
```diff
@@ -260,26 +260,68 @@ Plain widgets render all contents of an object or list sequentially without any
...
 ResourceList widgets render a list of Kubernetes resources. The ResourceList widgets should be used along with [related resources](resources.md#datasources-section).
 
-If such resource list was already defined in Busola, the configuration will be reused. To obtain custom columns, specify the `columns` field.
+#### Widget-specific parameters
 
-#### Example
+- **children** optional field used to obtain custom columns. If not set, the configuration will be reused based on the existing resource list defined in Busola.
+- **sort** - optional sort option. Array of objects with the following properties that allow to sort by it's value:
+  - **path** - _[required]_ contains the path to the data used for the column.
+  - **default** - optional flag. If set to true, the list view is sorted by this value by default.
```
Comment (grego952) [inline docs/extensibility/display-widgets.md:268]:
```suggestion
  - **default** - optional flag. If set to `true`, the list view is sorted by this value by default.
```

File: docs/extensibility/display-widgets.md [hunk 2]
```diff
@@ -260,26 +260,68 @@ Plain widgets render all contents of an object or list sequentially without any
...
+  - **compareFunction** - optional [JSONata](https://docs.jsonata.org/overview.html) compare function. It is required to use `$first` and `$second` variables when comparing two values. There is a special custom function `$compareStrings` used to compare two strings e.g. `$compareStrings($first, $second)`
```
Comment (grego952) [inline docs/extensibility/display-widgets.md:269]:
```suggestion
  - **compareFunction** - optional [JSONata](https://docs.jsonata.org/overview.html) compare function. It is required to use `$first` and `$second` variables when comparing two values. There is a special custom function `$compareStrings` used to compare two strings, for example, `$compareStrings($first, $second)`
```

File: docs/extensibility/display-widgets.md [hunk 3]
```diff
@@ -307,15 +349,24 @@ Table widgets display array data as rows of a table instead of free-standing com
 #### Widget-specific parameters
 
 - **collapsible** - an optional array of extra widgets to display as an extra collapsible section. Uses the same format as the **children** parameter.
+- **sort** - optional sort option. If set to true, it allows to sort via this value. Defaults to false. It can also be set to an object with the following properties:
+  - **default** - optional flag. If set to true, the list view is sorted by this value by default.
+  - **compareFunction** - optional [JSONata](https://docs.jsonata.org/overview.html) compare function. It is required to use `$first` and `$second` variables when comparing two values. There is a special custom function `$compareStrings` used to compare two strings e.g. `$compareStrings($first, $second)`
```
Comment (grego952) [inline docs/extensibility/display-widgets.md:354]:
```suggestion
  - **compareFunction** - optional [JSONata](https://docs.jsonata.org/overview.html) compare function. It is required to use `$first` and `$second` variables when comparing two values. There is a special custom function `$compareStrings` used to compare two strings, for example, `$compareStrings($first, $second)`
```

**Review by akucharska — COMMENTED** [review]
Other than inconsistence in configuration: once `sort` is an array, once object or boolean, I don't have any more questions

**Review by akucharska — APPROVED** [review]

File: docs/extensibility/display-widgets.md [hunk 4]
```diff
@@ -304,28 +304,71 @@ Plain widgets render all contents of an object or list sequentially without any
 
 ResourceList widgets render a list of Kubernetes resources. The ResourceList widgets should be used along with [related resources](resources.md#datasources-section).
 
-If such resource list was already defined in Busola, the configuration will be reused. To obtain custom columns, specify the `columns` field.
+#### Widget-specific parameters
+
+- **children** optional field used to obtain custom columns. If not set, the configuration is reused based on the existing resource list defined in Busola.
+- **sort** - optional sort option. It's an array of objects that allow you to sort by the value from the given **source**.
```
Comment (grego952) [inline docs/extensibility/display-widgets.md:310]:
```suggestion
- **sort** - optional sort option. It's an array of objects that allows you to sort by the value from the given **source**.
```

File: docs/extensibility/display-widgets.md [hunk 5]
```diff
@@ -304,28 +304,71 @@ Plain widgets render all contents of an object or list sequentially without any
...
+  - **source** - _[required]_ contains a [JSONata](https://docs.jsonata.org/overview.html) expression used to fetch data for the column. In its simplest form it's just the path to the value.
```
Comment (grego952) [inline docs/extensibility/display-widgets.md:311]:
```suggestion
  - **source** - _[required]_ contains a [JSONata](https://docs.jsonata.org/overview.html) expression used to fetch data for the column. In its simplest form, it's the path to the value.
```

File: docs/extensibility/jsonata.md [hunk 6]
```diff
@@ -36,3 +37,33 @@ Example from dataSources
...
+This function can be used to sort two strings alphabetically.
+
+### Function parameters
+
+- **first** - first string to compare.
+- **second** - second string to compare.
+
+### Example
+
+Example from the ResourceList widget
```
Comment (grego952) [inline docs/extensibility/jsonata.md:52]:
```suggestion
Example from the ResourceList widget.
```

**Review by grego952 — APPROVED** [review]

**Review by akucharska — APPROVED** [review]

**Pull request merged**